### ISSUE TITLE
Wrong password

### ISSUE THREAD

**Issue by IamConfused** [ethereum/mist#2411]
#### System information
``````
Version: `0.8.9`
OS & Version: osx 10.11.6
Node type: `geth(default)`
``````
Hi,
I created an Ethereum main account some weeks ago. 
The day I created my account the sync was surprisingly slow so I closed the program and started it again. Tried to sync for more than 24hours and it still did not finish the sync. So I decided to backup everything and after that I executed these steps:

mkdir /Volumes/Drive2/Ethereum
cp -rpv ~/Library/Ethereum/chaindata /Volumes/Drive2/Ethereum/
mv ~/Library/Ethereum/chaindata ~/Library/Ethereum/OldChaindata_to_delete_when_tested
ln -s /Volumes/Drive2/Ethereum/chaindata ~/Library/Ethereum/chaindata
rm -rf ~/Library/Ethereum/OldChaindata_to_delete_when_tested

My sync finally worked and my main account shows the amount of ETH I transferred to it.
BUT: Now I wanted to send some ETH. I was asked for my password and I used exactly the one I wrote down when I created my account but it always says: "wrong password". Same problem if I try to create a wallet contract. I typed my password at least 10 times now and I am 100% sure it is the one I defined.

I still have my keystore backed up. Will I be able to restore my account with this information?

**Comment by RomanRoss** [comment]
Hi,
I join to the topic. There is very similar issue.  After I transfered the wallet to my another computer, the password is not valid. I can see my balance, wallet is synchronizing, but can't I withdraw funds because my password is permanently wrong now. Is there some way how to solve this issue?
---
Ethereum-Wallet-linux64-0-8-10.deb
Ubuntu 16.04.

**Comment by RomanRoss** [comment]
I have solved my issue simply re-install the wallet. Password works fine now.

**Comment by evertonfraga** [comment]
@IamConfused as I can see that you are tech-savvy, would you please try to run geth on the console and perform those tests on the command line? Here's how:

1. run `geth console`, or `geth attach` if it's already started.
2. on geth console, execute the following command: `> personal.unlockAccount('0xACCOUNT_ADDRESS')`

Now it should prompt you for the password.
`Unlock account [0xACCOUNT_ADDRESS]
Passphrase: [INSERT PASSWORD]
true`


If the return of it is true, you've got your password safe, and this should be some RPC problem. Otherwise, we should dig this deeper. Thanks for reporting.

**Comment by IamConfused** [comment]
Dear Ev,

Thanks a lot for your detailed answer. It finally worked. I still don't know what exactly the problem was. Anyway I appreciate a lot that you took the time to answer on my comment!

**Comment by seesix** [comment]
I have the same problem I have backed up the my UTC key from my mining gig to a USB stick. I then installed the latest version ethereumwallet onto a second machine. My internet connect here in NZ is slow so it then took several days to sync geth.  

Once geth had finally synced I copied the key into the keystore on the newly synced machine.
My balance is displayed but  I am unable to transfer or create contracts due to incorrect password.

I am sure the password is correct as I only ever use the 1 password for everything.

Ubuntu 16.04
Geth Version: 1.6.5-stable
Ethereum Wallet and Mist v0.8.10

**Comment by smartins** [comment]
Same here, when I tried to transfer ETH out of the address I was asked for a password but it keeps saying it's incorrect!

**Comment by Lucianovd** [comment]
Hey guys,

Similar problem over here. Created a wallet a week ago, syncing took a while and I transferred some ether to my 'main account wallet'. After restarting my computer a few times and waiting for a while the syncing completed and there it was, the ether arrived. In the meanwhile (being new to the wallet) I created 3-4 wallets. None of those where asked for a password. 

This morning I installed mist next to the Ethereum wallet I already have (I don't know if it is possible that this has something to do with the problem). Also I changed the main wallet name a few times. 

Now when I try to send my ether it says wrong password, I am 99% sure I am not writing the wrong password. I tried reinstalling both of the applications, didn't work. Also I tried to open wallet with etherwallet and also here it says wrong password.

In the application one time I had an error message when I tried to get to my wallets, I don't know the error what it said but it had something to do with geth..

I really hope someone could help me fix this problem, or have any idea what it could be (next to mistyping the password :P)

Thank youuuu

**Comment by mdcrews5** [comment]
Why is this closed? A TON of people are having this same problem and there is NO solution!! I would think the devs would have some way or care to rectify this HUGE issue. We are NOT forgetting our passwords...they ARE NOT working!!!

**Comment by Spartakkus** [comment]
Same incorrect password issue. I'm sure it's the correct, but impossible to proceed. exported keyfile to MEW and also password error. It has to be a bug when setting password in ethereum wallet. this is very frustrating

**Comment by mdcrews5** [comment]
agree, this is strait BS, I wrote my PW down as soon as I entered it and i use it all the time. I am certain that it is correct.

**Comment by Hyperiun** [comment]
I see that not only the problem, I can not move the funds, I have MIST the last version, and the wrong password.

Anyone can develop a solution?
Thanks

**Comment by calvinke8** [comment]
I am also having this problem. Sure I did not forget the password as I wrote it down and transferred before. Has been saying wrong password for the past 2 weeks. All threads say we forgot the password but this does not seem to be the case with so many people.

**Comment by Hyperiun** [comment]
Thanks for your answer, I have already solved the problem.
The Mist 0.9 version does not work on my mac, I downloaded the 0.8 again and everything is correct, it does not connect the last server.

**Comment by mdcrews5** [comment]
How can you tell if it's fully connected to the server?

**Comment by Hyperiun** [comment]
Go to the Mist console

**Comment by frufru99** [comment]
same problem here, i created an account a month ago, used pwgen for the password, copypasted it to the wallet and to a txt file stored on a pendrive so 0 chance for mistake but the wallet keeps telling me now that the password is wrong

**Comment by calvinke8** [comment]
Same. Still no solution. Had password copied. Also found it on key logger and same password. Tried reinstalling. No luck

**Comment by rriot** [comment]
I am having the same issue. My password works fine on my other addresses in the wallet, but it won't work in the main Etherbase address. I really hope there is a solution to this. I have already transferred tokens to the Etherbase address, assuming the password I set would work. I tried unlocking the address through geth attach, but no go. Also, MyEtherWallet says "wrong passphrase" too when I upload the keystore file. When I upload the other keystore files to MEW, they unlock instantly using the same password. Anyone find a solution? I am going to try to install an earlier version of mist as suggested above, but I can't see how this would fix the issue as the address cannot be unlocked using MEW.

**Comment by pmt210** [comment]
any resolution to this issue? I am experiencing the same thing.

**Comment by grizwold2016** [comment]
I have exactly the same issue...I have no problem with main account....but Account 1 will not let me transfer...keeps saying wrong password.....I only use one password EVER! I tried to get password recovery and they have tried twice and cannot unlock...PLEASE HELP!!!

**Comment by lucas-iao** [comment]
same issue- transferred funds from main account in ether mist mac wallet to a 2nd account created within the wallet.  The original password and main account are fine. However, all the ether transferred to the 2nd account is stuck. I see it-but every time i try to send it, i receive "wrong password" . Any help Any ideas Any solutions?

**Comment by grizwold2016** [comment]
I have tried for almost one year to correct the same problem.....

Troy Marshall
Cell: 604.414.5395
Email: tmarshall@sd47.bc.ca

**Comment by sworteu** [comment]
I have this issue since december 2016 (after creating the wallet) it cointains alot of ethereum but i can't seem to access it. "Password incorrect". 

Tried CAPS LOCK on/off still no luck. Also Mist is missing a "Caps Lock is on" Warning.

**Comment by TGM** [comment]
Same problem here, created my walled using GUI, on version 0.8.10, not it just returns password incorrect, all my other accounts are working properly except one.

**Comment by 13enCoin** [comment]
Can confirm i currently have allot of eth sat in a mist wallet i created yesterday, and all its saying is 'Wrong Password', Uploaded the key file to myetherwallet and tried the password and once again 'wrong person' wtf is this crap, nearly all my money is locked in this account

**Comment by smartins** [comment]
I was never able to transfer funds out from one of my wallets (the default one created when installing the wallet), the password I used says it's not valid :/

**Comment by muax3000** [comment]
Same problem here. I have 2 accounts on the same wallet and used the keystore file with MyEtherWallet.com, but its coming up with an error - "wrong password".

**Comment by LubenCZ** [comment]
Hello, 

I´m having similar problem. Invested money to mining equipment and spend a lot of time by setting everything up. Now, after 1st ETH mined and synced with ethereum wallet I can´t send transaction. 

It say´s WRONG PASSWORD, however password is 100% correct since I saved passphrase to KeePass database when creating main account.

The only thing I did is that I copy\pasted jason file to another computer and installed wallet also there.

This is very annoying and I hope that someone from development team will take responsibility for this issue. It´s very unprofessional to leave people without solution while they giving trust and support this project.

Best Regards,
Michal

**Comment by 13enCoin** [comment]
Yeah tbh once downloading the official Ethereum wallet i just expected everything to work because it involves peoples money and its connected with Ethereum, i didn't expect it to just eat my money like it did.

My kids are going to have a worst Christmas because of a issue which shouldn't of existed in the first place caused by lousy programming. That money is now sitting in a Ethereum wallet which i can view open in my wallet but can't move/transfer. 

I've used countless wallets for various coins and never had a issue, then when i use Ethereums which you would expected to be one of the best it just eats my money? ... speechless.

**Comment by Madebyivan** [comment]
AND ABOVE CLOSING IT, THEY DO NOT HAVE SHAME!!!

**Comment by rriot** [comment]
I've had this issue for five or six months now, no resolution.

**Comment by Madebyivan** [comment]
Let's make it public, that they see that the coin falls, you will see how the batteries are put when we touch your pocket, it is clear that ours does not matter for what I see ...

We have to unite all those affected and create a group and make it public and make it reach the market, that people do not put their money, pq will lose everything.

We demand that we can move the money from one account to another without asking for a password if you are on the same computer.

It's humiliating because when he puts in he does not ask you for anything

**Comment by Ruben38** [comment]
Same problem here with one of my wallets for over an year now. I made a typo with my other wallet and found the password with a recovery program. If i can help anyone with the recovery let me know. I know how much it sucks to see your money sitting in an wallet and cant spend it >.<

**Comment by horyd** [comment]
I just created a wallet using Ethereum Wallet now too, and trying to import into MetaMask/My Ether Wallet is giving the same issue regarding passphrase being incorrect. I generated and saved it through 1Password. What on earth is going on?

**Comment by evertonfraga** [comment]
Hi all, 

I am reopening this issue, as I see it's not fixed yet. The team has grown since June and now we can breathe again. We'll keep you posted about fixes.

**Comment by LubenCZ** [comment]
Thanks a lot evertonfraga, hope you will find solution soon!

Best Regards,
Michal

**Comment by smartins** [comment]
@Ruben38 Can you share how you were able to recover the password? I'm in the same boat and would like to give it a try! Thanks!

**Comment by Creator77** [comment]
One of possible solutions:
https://github.com/ethereum/mist/issues/3176#issuecomment-351879937
Hope it will be helpfull for develpers too.

**Comment by LubenCZ** [comment]
@Creator77: actually I have dot special character "." at the end of my password as well, what I should do to type password correctly so dot is accepted?

**Comment by pavneet09** [comment]
Hi @evertonfraga 

Thank you for re-opening the issue. 
I too am facing this problem on 0.9.2 as well as 0.9.3.

The Main Account works fine with the same password, whereas the second account (Account 2) having the same password [Photo was taken while creating the account, Password backed up etc] does not work. 
The second account was created using Mist itself and not in the cmd like the main one. Both have special characters in them and it seems most people with special characters have an issue.

Thank you for continually trying to solve it. I would be highly obliged if you could let me know of any hit and trial methods that can be done to access the account. Regards.

**Comment by Ruben38** [comment]
Heya!
I found it with ethcracker. Check this link.
https://github.com/lexansoft/ethcracker

If you can’t figure it out. Let me know and i will give it a try for you.




Op 14 dec. 2017, om 10:53 heeft Steven Martins <notifications@github.com<mailto:notifications@github.com>> het volgende geschreven:


@Ruben38<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fruben38&data=02%7C01%7C%7Cce1f00c6ec6348b31ea308d542d87fba%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636488420012331101&sdata=e3X3sH89eTGm5A6TN5ivLneC3h3GdhI8WfPNKt%2FKasU%3D&reserved=0> Can you share how you were able to recover the password? I'm in the same boat and would like to give it a try! Thanks!

**Comment by Creator77** [comment]
@LubenCZ 
In our case dot was ignored too. Just try to remove the dot from your password.

And read connected issue:
#3176 
More info about special symbols is there.

**Comment by pavneet09** [comment]
@Creator77 

Hi,

I have read the thread before, but was there a solution found to the special character reference? I too have plenty of special characters in my passwrod including @ etc. Im guessing this is what causes the problem in the second account?

**Comment by LubenCZ** [comment]
@Creator77: I was trying send transaction using password without special characters, but issue remains unfortunately, in my case the root cause has to be somewhere else.

**Comment by Ios8162** [comment]
I store 72 passwords in the same way.  71 of them always work without fail.  My Etherium password has never worked.  I am 100% sure I did not make a user error and I have not been able to get my ETH out of my main account since February.  I have been keeping the keystore file since then hoping for a fix.  I have tried all suggestions I have ever seen including Geth commands, ethcracker, pythrecover, different versions of mist / geth / wallet, myetherwallet.com and probably a few I forgot.  

Like others, I also used special characters in the password and used a copy/paste command to input it into the password field.  Would love some follow up on this.  There is no way so many people are making user errors with their passphrases.

**Comment by JamesDall** [comment]
Am having the same problem with @ symbols in my case, 3 wallets. Used mist / Ethereum-Wallet 8.10, possibly (0.9.0)

Anyone know what the @ symbol turns into?

Have been looking into this issue for a few hours now: So far I've seen reports of:
. removing dot, and for some ignoring all characters after
^ and symbols like it, turned into ^, and/or missing, and next character missing
" missing and next character missing

This seems to suggest it has something to do with international symbol conversion. As in "ee -> ëe, "re -> e, type of behaviour.

Similar issues are reported here:
https://github.com/ethereum/mist/issues/3403
https://github.com/ethereum/mist/issues/3176

Anyone any leads?

**Comment by JamesDall** [comment]
Any ideas on the @ symbol, @pavneet09 or @andirayo?

**Comment by anormore** [comment]
Can some one confirm that @ may prevent my PreSale wallet from being opened?! My password ABSOLUTELY has this character in it. **Confirm Please**? So I can feel less stupid?

**Comment by pavneet09** [comment]
@JamesDall 

Hi, 

Yes, Im currently in the same boat as well with plenty of Eth locked up due to special characters used in the password "@" being one of them.

I've tried all possible combinations as of now. I took a picture while creating the account of the password as mist gives you the option of viewing your password while creating. I just never thought something like this would happen as it never happened with main account.

Currently waiting reply on many different platforms if something is being done to address this issue or we're going to be left at the mercy of third party apps to try and play around with millions of combinations to unlock said account.

The good news is @evertonfraga did recognize this as an issue and re-opened the case. Hopefully something will come out of it soon.

**Comment by JamesDall** [comment]
Good that we have found each other. Here are some more cases: https://github.com/ethereum/mist/issues/982

Also @larandi, @fireandthevoid, any ideas on the @ or other symbols in Ethereum passwords? And @abonn, did your password cointain @ or other symbols?

**Comment by mdcrews5** [comment]
I have an @ symbol in mine and it works. It is most likely a mistake
somewhere else in the password. An extra number you forgot, a different
combination, etc.

**Comment by JamesDall** [comment]
In 3 separate passwords, two of which I wrote down carefully before entering? Also, when did you make your password @mdcrews5? As with the current version & 8.1.0 version I could not re-create it either, so it must be something somewhat unusual.

Also see @elpollitodiablo's case, who did re-create it for the 'ß' symbol:
https://github.com/ethereum/mist/issues/3176#issuecomment-349403728

geth attach
personal.newAccount()
use passphrase FusRoDah
0xAccountA is created (replace 0xAccountA with hash)

personal.newAccount()
use passphrase FußRoDah
0xAccountB is created (replace 0xAccountB with hash)

Run this:

personal.unlockAccount('0xAccountA', 'FusRoDah')
true

and then

personal.unlockAccount('0xAccountB', 'FußRoDah')
Error: could not decrypt key with given passphrase

**Comment by mdcrews5** [comment]
I made mine back in July and could't figure it out forever.....wrote it
down and everything. For whatever reason, I used a similar password that I
usually use but left out 4 numbers which I always put in. for example
pass2017word.....I ended up not putting the 2017 like I usually did and
once I figured that out....it worked. For months, I was dead set that it
was a software issue. Turned out I was just a dumbass. Not saying that you
are or your situation is similar at all. Just throwing my experience in the
mix. Try out some different combos, you never know. I think there should be
a way to recover passwords for kids with dementia like myself, lol.

**Comment by JamesDall** [comment]
Two of these passwords were passwords I had never used before, consisting of common words, written down specifically to be unique, even transcribed on three separate pieces of paper (with print-out of the json as cold wallets).

So pretty unlikely that for those two some numbers or additional words would have snuck in. Also have already tried over 50.000 combinations of various misspellings, caps, etc. using a python script (that I also tested with a newly created wallet). Had already given up on the first of the 3 some months ago, but then I tried to transfer money from my main cold wallet this week, and was shocked to see it did not work either! Same for the final one. Then did some digging and found the issues other people have raised. Must be something structural going on here!

**Comment by mdcrews5** [comment]
Totally understand, could be any number of issues. I've also read a ton of
horror stories so you could definitely be correct. Wish you luck!

**Comment by JamesDall** [comment]
Thanks for the well-wishes @mdcrews5!

In the meantime I have found some more potential cases:
https://github.com/ethereum/mist/issues/2077
https://github.com/ethereum/mist/issues/982
https://github.com/ethereum/mist/issues/669
https://github.com/ethereum/mist/issues/637
https://github.com/ethereum/mist/issues/570#issuecomment-217430786
https://github.com/ethereum/mist/issues/3403
https://github.com/ethereum/mist/issues/3354

@Haxtro, @diegopau, @archywilhes, @sdiman, @ihatecreatingaccounts, @supere989, @226536k, @Neurolicious, @kooweele, @Foxcor, @kooweele you any idea on how symbols such as @ translate in/to passwords set with previous versions of mist?

@ursul0 do you still happen to have that VM of your original set-up? Might help with tracking down what the issue is!

@RostSP, what symbols were changed by word?

**Comment by JamesDall** [comment]
And some more:
https://github.com/ethereum/mist/issues/3342 (odd case of crashing with the @ symbol in pw)
https://github.com/ethereum/mist/issues/3328
https://github.com/ethereum/mist/issues/3244
https://github.com/ethereum/mist/issues/3223

@kooweele, did you find a solution?

**Comment by abonn** [comment]
Hi everyone. I had this problem until I downgraded to version 8. I also
meticulously write down my passwords and know that it was not working with
the newest version until I downgraded. I even tried exporting the PK and
using MEW to open it to no avail. It was not until I was able to change the
password in the downgraded version within the console commands that I could
access my wallet.

I will not be using this software anymore and instead have opted to utilize
other services, lest I have this problem again.

On another note, the only symbol I had in my password was a "?".

Best of luck to those of you still struggling...

**Comment by supere989** [comment]
​You issue is related to the current derivation state of the private key.
You key needs to be upgraded to the current derivation state to be
decrypted with your current password.​
There is a process for it, but I don't have the time to do it, nor the
incentive. I believe it was documented several months ago.

Thank You!

Raymond Johnson
509-521-8751
supere989@gmail.com

**Comment by AlexLevshin** [comment]
Guys, same problem.

**Comment by JamesDall** [comment]
@abonn, is that 8 as in 0.8.0 from July '16, or last with 8 in it, such as 0.8.10, last May? As downgrading to 0.8.10 did not work for me (did not fix the problem).

**Comment by abonn** [comment]
@JamesDall I downgraded to 0.8.9. I'm not certain, but I think you should try downgrading to the version you had when originally creating the wallet.

**Comment by JamesDall** [comment]
@abonn, just tried downgrading to 0.8.9. But no luck. Also mist automatically downloads the latest geth, and geth is the command-line tool. So it seems strange that downgrading Mist would help change passwords.

My hypothesis so far is that mist, probably some earlier version (< 0.9.0, though could be the current as well) mangled passwords that were input before sending them to geth for account creation. Thus passwords cannot be unlocked with the geth command line tools when typed directly. This does not happen always, but only with certain libraries / show password selected / copy-pasting / or keyboard or international OS settings.

**Comment by abonn** [comment]
@JamesDall  I have geth version 1.6.7 installed. Is this the version you used with the downgraded version?

**Comment by JamesDall** [comment]
@abonn, yes, have tried unlocking with that geth version.

**Comment by rriot** [comment]
I have tried unlocking with geth too with no results.

**Comment by JamesDall** [comment]
Here's some more cases, these are in Geth:
https://github.com/ethereum/go-ethereum/issues/15633
https://github.com/ethereum/go-ethereum/issues/14591
https://github.com/ethereum/go-ethereum/issues/2908
https://github.com/ethereum/go-ethereum/pull/2369
https://github.com/ethereum/go-ethereum/issues/2138 (something suggesting geth inside docker misbehaves, suggesting an environmental factor on how passwords are read)

On reddit:
https://www.reddit.com/r/ethereum/comments/4na19x/mist_wallet_password_problem/

**Comment by anormore** [comment]
**Does the invalid character affect other systems like MEW?**

**Comment by ursul0** [comment]
@JamesDall nope, no such backup exist anymore. bits are re-utilized for a better purpose :)

**Comment by FireAndTheVoid** [comment]
@JamesDall, you were asking about issues with special characters.  Here is what I know.

The Mist wallet is uses HTML for the user interface.  So, the field in which you type your password is a character-masked HTML password field.  It appears that some operating systems in combination with some installed keyboards (for example, a German language keyboard), may change the characters entered into the masked password field.  Because the password field is masked, you might not realize that a character that you entered was actually ignored or changed to another character.  The character that I have observed as having this issue was the ^ character.  It would simply be ignored by Mist.  Furthermore, even uninstalling the keyboard did not fix this issue.  BTW, this issue DOES NOT exist with the unmasked HTML text field, so you cannot observe it directly (unmasked).

Because of this issue, the following scenario is a possibility.  Someone might create their password with symbols using the native English keyboard.  Subsequently, they install another language keyboard.  The new keyboard causes the HTML password field to not recognize one of the symbols used in their password.  Even when typing in their exact password, they cannot access their account.  Suspecting that their installed keyboard is the issue, they uninstall it.  However, even when uninstalled, the behavior persists and the character is still not recognized by the HTML password field.

Another possible scenario is that you create your account, a symbol is ignored, and then you cannot unlock your account using the geth terminal with personal.unlockAccount().

I'm not saying this is the problem you are having, however, it is an issue that I am 100% certain exists.  Note that I have not observed this issue with alphanumeric characters.  They seem to be consistently treated correctly in the password field.  I suspect that the problem arises because of the ill-defined behavior of the HTML password field.  Some operating systems, keyboards, and whatever else seem to implement it differently.  Side Note:  In my own extensive testing on the Windows 10 operating system with the native English keyboard, I was unable to break Mist by making various passwords with different symbol, number, letter combinations.

I would recommend trying to recreate the environment from when you initially created your wallet, i.e. the same version operating system, the same keyboard, the same version of Mist, etc.

If you are still having trouble, you can e-mail me at fir3andth3void@gmail.com.  I've written a few tools to debug these wallet problems.

**Comment by nfkb** [comment]
I've got this password issue too. I've tried to unlock my wallet with ethcracker and tried thousands of variations of my usual password without success :(

**Comment by IvanM1993** [comment]
hi. I also tried ethcraker and through the geth console, and all without success. there are options yet, but haven't tried. my password contains symbols _, may be because of this character swears the Mist.

**Comment by pavneet09** [comment]
Have been playing around with ethcracker as well as a couple of scripts I wrote myself to try all possible variables. I'm kind of still looking from any sign that if this is a real issue or not from the developers POV and if anything is being done to address it.

I know considering the market, a lot of work must be being done to stabilize the network and add security etc and this might be a low priority issue, but some timeline would really be appreciated. For now it all seems very bleak :/

**Comment by anormore** [comment]
"I would recommend trying to recreate the environment from when you initially created your wallet, i.e. the same version operating system, the same keyboard, the same version of Mist, etc."

Some of us are stuck from the PreSale wallet. So that means we created our wallets from Ethereum.org website. We cannot re-create the environment. 

We cannot do anything.

**Comment by anormore** [comment]
@obscuren This issue needs more attention, or at least a statement from Ethereum

**Comment by anormore** [comment]
@vbuterin  This issue needs more attention

**Comment by joshbarraza** [comment]
Thanks for helping spread the word @anormore 
I, too, cannot get into my wallet. Honestly, I don't even recall setting a password. The password I would've used has special characters like @ and ) and - 
I hope to regain access to my wallet at some point. A few weeks ago I thought all hope was lost.

**Comment by anormore** [comment]
Yep. I hear you buddy. I spent a lot of money cracking for my password, and it turns out, it's Ethereums fault. I'm pretty pissed. What's more, I used a third party to assist, so now they have my wallet.json and password list. So as soon as Ethereum fixes this problem, it'll be a race to steal my coins.

This is a huge embarrassment for Ethereum, and this list is growing.

**Comment by asyms** [comment]
Someone referenced my thread a couple of days ago, and I'm glad to find out you guys are still trying to get justice!
I don't know if there's anything else we can do beside trying to keep this thread alive and hoping a dev gives us a possible solution?

**Comment by KikoH** [comment]
Same issue here. No recollection of ever creating a password for my second wallet. I really hope a dev addresses this.

**Comment by j12342** [comment]
Same Issue here. I can see my balance in Mist after taking forever to sync. Now I dont have enough space to sync anymore. 

I 100% wasn't asked to create a password. If I was/did I would have written it down. Have tried every possible combination of the passwords that I use. But I'm sure I wasn't given the opportunity to set one. I'm on an Macbook air

**Comment by Ruben38** [comment]
Same problem here. Can't get my eth out of my 2nd and 3rd wallet. Tried alot of the possible solutions nothing is working. ethcracker, downgrade it to previous version, use geth. Spend a lot of time to solve this, but i cant figure it out. Is there any update on this problem or a timeline?

**Comment by larandi** [comment]
Guys, had the same issue,  I was 100% sure I didn`t put any password, in July 2017,  but the reality is a human brain sometimes can forgot.  I hired a programmer to edit and fix the cracker for more options and better performance and spread it to a several Xeon servers, after a week of cracking I`ve found a password, then I squeezed into the forehead and said OMG, yes that password I could really use, is more secret, then ususally, but smells like one of mine :-) I`m just forgotting dumb.  So I was successful. But I made not a small  investment to programmers job and spent tons of my time. Then now I have a fuctional cracking solution, If you want to buy a working software or try my services, then contact me on laterian@seznam.cz , for a better trust , we can communicate via FB also.  My motivation is only a return of my investment not any fraud. Just fair play.

**Comment by AndyNormore** [comment]
BE CAREFUL of @larandi - It is very reasonable that soon Ethereum will fix the password issue, and you've gone ahead and given your key and passwords away. DO NOT give out your keys/passwords until official word from Ethereum. I'll check out Larandi's solution for now.

**Comment by asyms** [comment]
If @larandi is actually trying to help, he could maybe tell us what "extra options" he added to the cracker / how he improved it?
Any information might help!

**Comment by AndyNormore** [comment]
I just finished my email conversation with him, he's fishing for your Eth information. DON'T TRUST ANYONE with your stuff. He's using Pythrecover and hired a guy to modify it because he couldn't get it running. EthCracker and Pythrecover work just fine, I've recovered three wallets for clients (lucky them). Never use some one elses source code, download it from GitHub and run on your offline machine or you CAN be stolen from. 

If this wasn't an official bug, I would say go for it, trust some (as I have) to crack your wallet - but this IS INFACT an official Ethereum bug. Therefor, sit tight.

However, if the Ethereum.org team doesn't address this soon, I will put a petition together so we can voice our displeasure with this insane issue.

**Comment by larandi** [comment]
Yes, I`m trully a phisher,  :-) I do not want  your wallet if you want, just offering an improved script, with more options, set as a parameters.

**Comment by larandi** [comment]
Just one thing regarding password issue  , half year ago I had also a locked wallet, I was 100% sure, there is an issue. I remember I set no password to my wallet, but finally I saw, the password was there and was mine, slightly different, but 100% set by me. Since this moment I believe, there is an option there is no password issue, but only our brain issue.  Maybe I`m wrong, but I believe there will be someone with the same idea as me :-)

**Comment by asyms** [comment]
Why not post it here @larandi ?

**Comment by larandi** [comment]
np, parameters are point 15 in the manual


15) syntax ( parameters are explained below also)
python louskac.py [-h] -p PENEZENKA [-v VLAKEN] (-w WORDLIST | -b BRUTE)
                  [-d DELKA] [-k KONCOVYSOUBOR]

python louskac.py -h   you will see help
 where parameter -p means a wallet name ( wallet file must be in a folder louskac)  
parameter -w means a list of all possible passwords, part of passwords and words or symbols which you ever combined to create password.   A practical way is to sit down and type all the fragments, numbers or other symbols which you have ever used to create passwords. The program will start to combine it.  An example of words.txt is already enclosured. 
parameter -b is brute force and just generate all the possible symbols, you can put ascii for all the symbols or e.g. 012345abcjk, all the characters will be uded extra and combined ( 0, 1,2,3,4,5,a,b,c,j,k)
parameter -v means number of used threads of CPU
parameter -d means a lenght of the password , use eg -d 10 if you know the password is not longer than 10 characters
parameter -k means the password will always ends by word in a file ( e.g. look into file konec.txt or end.txt , here is used konec.txt , what is in the file you can see by midnight commander ( use F3 to open file ) )
examples of usage 

python louskac.py -p UTC--2016-03-23T16-37-34.229474800Z--92121a6739aba7031b15d50ce71123b9ac3be12d -w words.txt -b 0123abc -d 12 -k end.txt
it will combine all the password fragments from words.txt file with all combinations with chars 0,1,2,3,a,b,c , will skip all combinations longer than 12 chars and password will always end by words from the file end.txt
next examples
a)
python louskac.py -p UTC--2016-03-23T16-37-34.229474800Z--92121a6739aba7031b15d50ce71123b9ac3be12d -w words.txt
b)
python louskac.py -p UTC--2016-03-23T16-37-34.229474800Z--92121a6739aba7031b15d50ce71123b9ac3be12d -b ascii -d 1 -k konec.txt 

c)
python louskac.py -p UTC--2016-03-23T16-37-34.229474800Z--92121a6739aba7031b15d50ce71123b9ac3be12d -b t -d 15 -k konec.txt
d)
python louskac.py -p UTC--2016-03-23T16-37-34.229474800Z--92121a6739aba7031b15d50ce71123b9ac3be12d -w wordlisty/words_01.txt 
here the file words_01.txt is in folder wordlisty

e)
python louskac.py -w wordlisty/words_01.txt -b 1234596789
 
this means program will combine all the words from word list words01.txt and together with numbers 1,2,3,4,5,6,7,8,9  , so all the symbols after -b are used and combined in all variations , you don`t need to make gaps or comma between






You can also generate a password list words.txt by 

python seqwl.py

**Comment by AndyNormore** [comment]
@larandi Thank you for contributing your solution to this problem, however, it is not relevant to this thread and bug. The topic here is the BUG from Ethereum regarding password special characters. It's a strange bug.

Feel free to post your services at http://reddit.com/r/ethereumlostpasswords, where the crack fee is 20%. There are certainly a percentage of Ethereum owners who have forgotten their passwords. 

This is not the case. Users here have substantial evidence of password **corruption** from the Ethereum wallet software, and we would like it corrected.

**Comment by VCBCN** [comment]
I’m not sure if this applies to the people with pre-sale wallets, but maybe it’ll help someone, or even point the devs in the right direction.  I was driving myself crazy thinking that I had forgotten the password, but didn’t remember setting one up.

 I came across an article somewhere about setting up the wallet, and it seems like there’s a “skip” option when it comes to setting up the password. This rang a bell for me and I know that that’s the option I chose. Is it possible that the people experiencing this bug are the people that clicked “skip”? So the wallet is requiring a password that we had chosen not to create?

**Comment by ursul0** [comment]
just to remind you of the issue I had and solved about a year ago:
1) I had wallet created in the geth console  (around the block 200000, password stored in password vault)
1.5) I also added another wallet, beside coinbase (password for this one ALWAYS worked fine) 
2) It was all fine until I tried Mist few months later where I discovered that the password won't work (it did worked in geth before as I was able to unlock)
3) I tried everything I could think of unsuccessfully until I brought back the backup of the original machine, where in geth console I was able to change the password (using the one I always thought is the right password) with "account update" command.

**Comment by pavneet09** [comment]
Hi all,

Here's hoping something good comes from this thread in 2018 but for now it all kind off seems unlikely. So recently I ran into another issue with Mist where while transferring some Eth I found out that the fees I was about to pay for the transfer was 1 ETH!
Obviously that had me sort of shocked, the least amount of fees I could pay was 0.5
Anyhow I didnt go through with it and after going through a few issues found out it was usually due to the light client of geth causing the issue and losing peers. 
However I did realize that @evertonfraga, you mentioned that organization was well aware of the problem as was looking towards fixing the same in the 0.9.4 version of Mist.
Which brings me back to the current issue, as we all sit here with the Eth just locked away in the wallets, the issue milestone has been regarded as 0.9.5 version. 

Does this mean that currently this is not being addressed in any way? Is there any support that the org requires from our end to help move the process along? 
Because sitting here with folded arms and waiting is just not cutting it. Again terribly sorry for the repeated bumping, but I'm really in a bind here myself.

**Comment by iYalovoy** [comment]
[Copy/paste issue may depend on keyboard layout.](https://github.com/electron/electron/issues/10060) Might be relevant.

**Comment by GRT314** [comment]
I have this problem to. I have a couple of passwords with ! and $ in them, now they dont work on expanse using Mist 0.8.9. Account  made with  mist 0.9.3 for etherum works with same password. 

I have tried to access the wallet in geth, in both versions of mist (copied wallet file into ethereum keystore) and on myetherwallet.
I did not copy/paste password but have norwegian keyboard layout.
I would guess the password is encoded wrong with special characer

**Comment by pavneet09** [comment]
Got it all in my password ha.

! @ # $ 

I think I'm gonna be here for a while!

**Comment by Joshwink28** [comment]
Hi guys, the same problem. I can see my ETH but because "wrong passwort" i cannot send. Im sure that i have right passwort, i wrote it down...

**Comment by kooweele** [comment]
If you have special chars in your passsword —- then it’s scrambled for good. I too have a screenshot of my password when I created it in Mist.


Sent from Yahoo Mail for iPhone


On Wednesday, January 3, 2018, 23:32, Joshwink28 <notifications@github.com> wrote:


Hi guys, the same problem. I can see my ETH but because "wrong passwort" i cannot send. Im sure that i have right passwort, i wrote it down...

**Comment by pavneet09** [comment]
@kooweele 

That isnt exactly the case. I have an image of my password as well, that I took while creating in Mist. This was rather the second account I created using Mist. 
The previous account was created using Geth itself and the same password with the same special characters is fine. 

There has to be a better explanation then, it just got scrambled up or the special characters got changed up during creation, because if thats the case, there needs to be a genuine search for a solution to this. Unfortunately it seems the percentage of people using Mist or any offline wallet is not that high, most of the volume must be consumed by coinbase/exchange based wallets so we might just be a tiny percentage of people stuck with no access to the token and hence it doesn't seem too high a priority atm.

**Comment by anormore** [comment]
I created my wallet from the PreSale from Ethereum.org, and have spent about a grand recovering my password. Better be worth it, I have 250 Eth stuck.

**Comment by archywilhes** [comment]
It's nice to see people discussing about this problem again. In my case it was a password of 400+ length with a lot of special chars. And funny enough I created 2 wallets with the same password (both via Mist) and I am able to decrypt the encrypted private key for the first wallet but not the second one [Though I'm pretty sure they share the same password that I copied & pasted into the HTML input element in Mist]. 

In my case there are basically 2 ways that a bug could arise that would result in the problem of password mismatch:

1. Special char got converted into something weird in the HTML input element before they got fed into `geth`
2. strange `geth` bug with aes where they gave you the wrong salt or iv or something after 2 encryptions of the same string over a short period of time.

More than half a year have passed and I had tried brute forcing with possible special char mutations, tinkering with the salt & iv, looking into Mist's code to see if I could find any clue in a possible fuck-up in the process of feeding the string from GUI to `geth` (well there is really nothing unusual going on. just a normal Meteor app.) and I still couldn't decrypt the private key and get my ethers back.

**Comment by sebd-davra** [comment]
I have the same password issue with a just new created  ethereum wallet with mist, my password has been generated with keepass with only alphanumeric characters, and my keyboard selected in windows is ENG INTL and I copied/pasted the password.

**Comment by AndyNormore** [comment]
@evertonfraga We are still waiting for an update what is happening with this?!

**Comment by pavneet09** [comment]
Hi everyone, please refer to @marcgarreau 's new thread and try and provide as much as substantial help as possible.

He has been trying to replicate the problem for a while to solve it. It will be better for everyone if we contribute as much as possible. 

See you all over at https://github.com/ethereum/mist/issues/3513

**Comment by Ios8162** [comment]
Excellent!  Glad to see this update.  I will jump over to the new thread and contribute to the bug reports.
#3513

**Comment by FireAndTheVoid** [comment]
To everyone that is 100% certain of their password and still cannot access their wallet, you should try to recreate the problem with a test account.  The problem needs to be reproducible in order for it to be debugged.

If the problem is deterministic and happens every time you create an account, then I should be able to help you (the devs can help too) figure out what's wrong.  If the problem is non-deterministic and happens some percentage of the time (e.g. 10% of the time), then it should also be possible to debug with a little more effort.

Remember, you don't need to use the same password as you used with your funded Mist account.  A similar password with the same special characters, same number of alphanumeric characters, etc should be sufficient.

**Comment by iYalovoy** [comment]
@FireAndTheVoid You also should take into account language issues. If you have more than one language installed at your system and copy-pasted pwd into mist, it might be turned into something  completely different. This issue is confirmed by Electron and Mist is written on Electron.

**Comment by sebd-davra** [comment]
I have more than one keyboard language installed ( english and english international) and I copy-pasted into mist do you know the transformation process it should be, what char is turned in another one ?

**Comment by iYalovoy** [comment]
@sebd-davra no I don't and it depends on your system set up. More here https://github.com/electron/electron/issues/10060

**Comment by KikoH** [comment]
So my issue happened on my second account when I upgraded geth through the popup when I started mist. I installed mist July 22nd so I am not 100% sure which geth version it came with then. Also not sure how to downgrade geth (in mist). If anyone can help me so I can try to debug and help in the other thread. I too know my password and it doesn't work on second account.

**Comment by sebd-davra** [comment]
Thanks iYalovoy that's left a little bit of hope.

**Comment by iYalovoy** [comment]
@sebd-davra there is actually a lot of hope. As times goes on ether value grows, it becomes more attractive task and there are more and more information on the issue. When it will cross a certain value I'll probably write permutation script in python for my own pwd in order to try all sequential permutations of my pwd including language mappings. If I make it, I'll make it open-source for sure. I think many people will do the dame.

**Comment by marcgarreau** [comment]
Closing this issue in favor of https://github.com/ethereum/mist/issues/3513. If you're able to reproduce any related bug, please help us out by documenting the detailed steps in that issue. Thanks!

**Comment by m-thomson** [comment]
Downgrading to an earlier version of geth (0.8.1) solved this problem for me and a few others. The password works again! Please try it folks. If you get access, immediately transfer your ETH to another address you control.

**Comment by sebd-davra** [comment]
@m-thomson geth 0.8.1 does not exist, you mean mist 0.8.1 or ethereum wallet 0.8.1 ?

**Comment by anormore** [comment]
Luck you, some of us are stuck on Pre-Sale wallets... so what version would we use?

**Comment by m-thomson** [comment]
@sebd-davra Yes, sorry I meant Eth Wallet 0.8.1 from https://github.com/ethereum/mist/releases/tag/v0.8.10


@anormore This was originally a pre-sale wallet as well. I don't know what version you should use. I only know that downgrading to 0.8.1 worked for me and I had the *exact* symptoms described in this thread.

**Comment by sebd-davra** [comment]
0.8.1 or 0.8.10 ? you give the link for the v0.8.10 not the v0.8.1

**Comment by m-thomson** [comment]
Sorry, I forgot version numbers don't work like decimals. I guess the trailing zero is important.

0.8.10 is the version I used (0.8.1 doesn't exist).

**Comment by sebd-davra** [comment]
the tagged version 0.8.1 still exist on github but was not released : https://github.com/ethereum/mist/tree/0.8.1

**Comment by m-thomson** [comment]
I'm positive that I downloaded it from the release page so that wouldn't be it.

**Comment by r3lax3d** [comment]
I am sorry to write in my role as IT Deliver Manager here. Not to abuse people and slow down the positive input here but this is a MAJOR issue that should have high priority and is OS type, Mist, Wallet and Geth version independent. Mist Version 9.0.2 also creates this issue. I am missing a taskforce here. High impact on future usage of Ethereum it's blockchain. Two years and no detailed test reports ? How many people are involved ? Why is this not on the agenda of the Ethereum Core Devs Meetings.

**Comment by ontheronix** [comment]
@abonn How did you manage to get such an old Mist version up and running? I tried all older versions but none boots completely.

**Comment by abonn** [comment]
@ontheronix I finally fixed this issue about two months ago so I cannot provide exact details. Someone posted a link to a github with a repository of the old versions. I downloaded this old version and then I believe opened up command prompt and utilized geth to update my main account in the program.

**Comment by oldmate89** [comment]
@m-thomson - Great to hear that you were able to resolve this issue :) 
I have been attempting to crack my pre-sale wallet for the last week with no such luck. 
Can you please confirm: 
1) That this was a presale wallet? 
2) Could you please outline which special characters you had in your password? 
3) Do you have multiple languages installed on your computer?

**Comment by r3lax3d** [comment]
@abonn 
Thats realy good news. A question about the version. Was it the initial version where you created the wallet / Main account with and that fixed it or did you used an even older version due to a link that was provided. Would realy like to know this.
Similar like the question as point 4 from @oldmate89

**Comment by IvanM1993** [comment]
hi all. The first version of the wallet I was 0.8.10, may 23, 2017. I have created a basic purse, before I had another wallet that I created on the website myetherwallet.com later I imported this wallet in the app, threw on a basic wallet a little ether. The password for the base of the purse was the same as for the purse, which was created on the website myetherwallet.com. later came the version 0.9.0 and I have updated the app. I wanted to send with the basic purse to another purse a little bit of ether, but the password was wrong. Checked basic wallet password on the website myetherwallet.com but there is also information about wrong password. another wallet with the same password worked. in January I switched to an initial version 0.8.10, but the password still didn't work for this wallet. When I created the wallet of the base in a line for a password I copied from the text editor the same password as for another purse, whether it could affect the work. this is the big question for me.

**Comment by pal-presaleWrongPW** [comment]
@FireAndTheVoid how would you re-produce a pre-sale wallet encryption?

**Comment by pal-presaleWrongPW** [comment]
@abonn how did you utilize geth to update your main account in the old version you installed?

**Comment by FireAndTheVoid** [comment]
@pal-presaleWrongPW Sorry, I've never worked with pre-sale wallets before.

**Comment by svobodnu1** [comment]
i can help recover password

**Comment by BreakHeadz** [comment]
Collect on new life.. seems lost password on forever
0 x FE902372f2F9561bb3CfFc5e191F9B7C9F2615a1 
Tnx who care. 2018 peace

**Comment by svobodnu1** [comment]
@BreakHeadz write me in the mail above and try to help you password pc-dopomoga @ mail. ru remove spaces

**Comment by teasider** [comment]
This issue still exists.
My locked wallet was created on 18th of may 2016.
I am syncing geth at the moment and downloaded the version of mist that fits that date to try and load the wallet there.

I only ever use 2 kinds of passwords, already tried hashcat with no results.

**Comment by sebd-davra** [comment]
Sure the issue still exist, it's a trap wallet.

**Comment by xander77a** [comment]
My ETH is trapped as well. I don't recall ever making a pw for this useless wallet. I'm on OSX and initially downloaded in late 2017 iirc. 

My questions are, what is the best brute force method, where is my keystore file, and can the attempts to crack this wallet originate from another lan connected device, or can the wallet be transferred to a Windows machine (isolated with substantially more power) and cracked there?

Lastly, have the ethereum developers acknowledged, or apologized, or resolved this horrible bug?

**Comment by Zhuvikin** [comment]
I found that `\` character is ignored.

My password contains a lot of special characters and it was failed to send any transactions for a several months from my Ethereum Wallet due to incorrect password error. Now, I just skip `\` (backslash) in my password and it works! 

Do not use backslash in your passwords!

**Comment by advancedsoftwarecanada** [comment]
INSANE. I hope this helps someone @Zhuvikin 

@evertonfraga PLEASE COMMENT.

**Comment by calvinke8** [comment]
damn I didnt have a backslash tho. rip

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by ihatecreatingaccounts** [comment]
(buy bitcoin) ;)

Am Do., 13. Juni 2019 um 22:38 Uhr schrieb calvinke8 <
notifications@github.com>:

**Comment by philsmd** [comment]
hey @Zhuvikin . It's an interesting problem you have found but are you also able to reliably reproduce this problem ? can you create a new wallet with a similar password containing the \ character and this problem is still the same and happens all the time ? which geth version, which mist version and which operating system do you use ? The operating system language and keyboard layout (language) might also be an interesting factor here... but it's still all only a guess and no real facts.

The best thing would be a step by step guide/tutorial on how to reproduce this and maybe even record your screen while trying to reproduce this, otherwise it's still not really "proven"/confirmed and might also be considered to just be a mistype on your part etc.

I think @holiman might be interested in this too, but we already saw a lot of this "claims" and nobody was reliably able to reproduce such issues i.e. it could also just be that s/he thought that s/he typed xyz while she typed xz instead (except the problem with erroneous trailing new lines and the liner problem in geth that I've reported)

**Comment by PhilippLgh** [comment]
Did a quick test with mist 0.10.0 by copy pasting `1234567\n89` from a text editor into the password input. The created geth keyfile can be unlocked outside of Mist with `1234567\n89` (if used as string in code you might want to write `'1234567\\n89'`). So all characters from the user input are also used in the password. No information is changed, lost or misrepresented. Though I see how it can be problematic to unlock such accounts in certain UIs that treat `\n` differently. I also agree that these kind of passwords should not exist in the first place but can also *NOT* reproduce the reported behavior. It would be important to get more details about the environment and versions being used or even the exact password.

**Comment by Zhuvikin** [comment]
Hi @PhilippLgh , I wasn't able to find out in logs my previous version of wallet I used when account was actually created. But I recorded a gif where I reproduce the issue. This is an account when I found the bug in. I will not use this account anymore, only for testing purposes. So, I can reveal the original password which was used for account creation: `H;bEUo#\BOkSez-.&KpY]o4A[[ozH0x`.

You can see that in geth I'm able to unlock account with both `H;bEUo#\BOkSez-.&KpY]o4A[[ozH0x` and `H;bEUo#BOkSez-.&KpY]o4A[[ozH0x`. However in wallet I can't transfer with the original version of the password. But it goes without slash.

![cRj9cKpTNP](https://user-images.githubusercontent.com/4532892/59981113-dee50800-9607-11e9-82f8-f2031110f19a.gif)

**Comment by Zhuvikin** [comment]
My current version of wallet is `0.11.1`
geth version: `1.8.27-stable`
OS: `macOS 10.14.5 (18F132)`

**Comment by holiman** [comment]
This is the case in both geth `console` and a browser (FF) that I just tested with:

```
> "a\Bc" == "aBc"
true
> "a\Cc" == "aCc"
true
> "a\Dc" == "aDc"
true

```
Apparently it's a javascript string thing. 

However, if the 'raw' mode is used, where we don't just take a javascript string, but actually the terminal input, we can accept the key sequence. I modified version geth to show what happens in the backend, with this diff
```diff
[user@work go-ethereum]$ git diff
diff --git a/accounts/keystore/passphrase.go b/accounts/keystore/passphrase.go
index 1ced41e99..af184cc81 100644
--- a/accounts/keystore/passphrase.go
+++ b/accounts/keystore/passphrase.go
@@ -45,6 +45,8 @@ import (
        "github.com/pborman/uuid"
        "golang.org/x/crypto/pbkdf2"
        "golang.org/x/crypto/scrypt"
+        "github.com/ethereum/go-ethereum/log"
+
 )
 
 const (
@@ -104,6 +106,7 @@ func StoreKey(dir, auth string, scryptN, scryptP int) (accounts.Account, error)
 }
 
 func (ks keyStorePassphrase) StoreKey(filename string, key *Key, auth string) error {
+        log.Info("storekey: password", "pw", auth, "pwhex", []byte(auth))
        keyjson, err := EncryptKey(key, auth, ks.scryptN, ks.scryptP)
        if err != nil {
                return err
@@ -138,6 +141,7 @@ func (ks keyStorePassphrase) JoinPath(filename string) string {
 
 // Encryptdata encrypts the data given as 'data' with the password 'auth'.
 func EncryptDataV3(data, auth []byte, scryptN, scryptP int) (CryptoJSON, error) {
+        log.Info("encrypt: password", "pw", auth, "pwhex", []byte(auth))
 
        salt := make([]byte, 32)
        if _, err := io.ReadFull(rand.Reader, salt); err != nil {
@@ -199,6 +203,7 @@ func EncryptKey(key *Key, auth string, scryptN, scryptP int) ([]byte, error) {
 
 // DecryptKey decrypts a key from a json blob, returning the private key itself.
 func DecryptKey(keyjson []byte, auth string) (*Key, error) {
+        log.Info("decrypt: password", "pw", auth, "pwhex", []byte(auth))
        // Parse the json into a simple map to fetch the key version
        m := make(map[string]interface{})
        if err := json.Unmarshal(keyjson, &m); err != nil {

```
So, using the terminal prompter, instead of a js-string:
```
> personal.newAccount()
Passphrase: 
Repeat passphrase: 
INFO [06-23|22:33:38.918] storekey: password                       pw=a\\Bc pwhex="[97 92 66 99]"
INFO [06-23|22:33:38.918] encrypt: password                        pw="[97 92 66 99]" pwhex="[97 92 66 99]"
INFO [06-23|22:33:39.671] decrypt: password                        pw=a\\Bc           pwhex="[97 92 66 99]"
INFO [06-23|22:33:40.467] Your new key was generated               address
```

```
> personal.unlockAccount("0x084E39F61973D2E39c14efCBd98EDdbda74Fe51e")
Unlock account 0x084E39F61973D2E39c14efCBd98EDdbda74Fe51e
Passphrase: 
INFO [06-23|22:34:25.825] decrypt: password                        pw=a\\Bc           pwhex="[97 92 66 99]"
true
```

Also, if I specify it in a password file, it also works:
```
[user@work go-ethereum]$ cat pw.txt 
a\Bc

[user@work go-ethereum]$ build/bin/geth --unlock 0x084E39F61973D2E39c14efCBd98EDdbda74Fe51e --password "pw.txt"
...

INFO [06-23|22:37:28.114] Unlocked account                         address=0x084E39F61973D2E39c14efCBd98EDdbda74Fe51e
```

Although this is quirky, and obviously a mismatch between geth CLI and Mist, it should be totally reproducable, and not subject to change across versions, and if you created the password in Mist it should still work in Mist. In practice, it's pretty easy to test a given string, if it's echoed back intact when you type it in a javascript console. 

```
> "test\A me"
"testA me"
```
In general, the javascript console is more of a dev-tool, and string escape sequences may be surprising to people who are not familiar with javascript. Consider: 
```
> console.log("test\n me \n foo")
test
 me 
 foo
undefined
```
So if someone has used "my\npassword" as a javascript string, that person may be surprised that the actual password contains a linebreak, and not literally `\n`

**Comment by Zhuvikin** [comment]
@holiman you're right. This was my fault. I did misunderstand that thing when was creating account several months ago. Just tried to unlock with password prompt.

```
> personal.unlockAccount(eth.accounts[0])
Unlock account 0xd49f78c91475c61d50388fc24896ac7d2c83ca39
Passphrase: 
(prompt pass with slash)
Error: could not decrypt key with given passphrase

> personal.unlockAccount(eth.accounts[0])
Unlock account 0xd49f78c91475c61d50388fc24896ac7d2c83ca39
Passphrase: 
(prompt pass without slash)
true
```
 
Should never use JavaScript strings to create new accounts and passwords! 😃

**Comment by chaev** [comment]
Finally, I solved this problem with my old account created in Ethereum wallet. I had ! in the password, and with it I always got "wrong password". Now I typed the password without ! and my keystore is unlocked! Try to play with special characters in your password.
